docs/DeveloperGuide.md

```diff
@@ -1,34 +1,216 @@
+- ``Quiz#quizNext()`` - Retrieves a random question from the available questions for that subject.
+- ``Quiz#markCorrectness()`` - Marks user answer as correct or wrong based on user's judgement.
+
+Given below is an example usage scenario and how the quiz mechanism behaves at each step.
+
+Step 1. Before the user decides to start a quiz, he/she has to create a subject using the command
+ ``addsubject s/SUBJECTNAME``, and add flashcards to that subject using the command ``addcard s/SUBJECTINDEX q/QUESTION a/ANSWER``.
+
+Step 2. The user can start a quiz by indicating a subject and the number of questions to quiz using the command
+``quiz s/SUBJECTINDEX n/NUMBERTOQUIZ``. If the number of questions to quiz is not specified, all the questions stored
+in that subject will be quizzed.
+The following diagram describes how the quiz operation works:
+
+![](images/quiz_sequence_uml.png)
+
+Step 3: The quiz will end upon completion of the specified number of questions, or by stopping the quiz using the
+command ``exitquiz``.
+
+#### 2.x.2. Design Considerations
+##### Aspect: How the user can control how many questions to be quizzed
+- **Alternative 1**: Always quiz all stored questions for that subject, but allow users to stop the quiz whenever they want.
+  - Pros: 
+    - Simple implementation.
+  - Cons: 
+    - The user will have to keep track of how many questions he has already done.
+  
+- **Alternative 2 (current choice)**: Allow the user to both set the number of questions to quiz and to stop the quiz halfway.
+  - Pros:
+    - More flexibility for the user.
+  - Cons:
+    - More complex implementation.
+    - The user has to type in a longer command to start a quiz, or they will be quizzed with all questions by default.
+    
+### 2.x. Score Feature
+#### 2.x.1. Implementation
+The score feature builds on the quiz feature, storing the score for each quiz session in a ScoreList. 
+This allows users to see all past scores and track any improvements. It implements the following operations:
+
+- ``Subject#showScores()`` - Shows all scores and the average score from all quiz sessions for that subject, in chronological order.
+- ``ScoreList#getAvg()`` - Calculates the average score for all scores in the ScoreList.
+- ``ScoreList#listScores()`` - Lists out all scores in the ScoreList.
+
+Given below is an example usage scenario and how the score mechanism behaves at each step.
+
+Step 1. Before the user can view his scores for a particular subject, he first needs to have done at least one quiz
+session for that subject using the command ``quiz s/SUBJECTINDEX n/NUMBERTOQUIZ``.
+
+Step 2. The user can view the score history and average score of a selected subject, if he has done at least one
+quiz session for that subject, using the command ``score s/SUBJECTINDEX``.
+The following diagram describes how the score operation works:
+
+![](images/scores_sequence_uml.png)
```

Comment:
If I remember correctly, the activation bar should extend all the way down for your parse(command)?

![image](https://user-images.githubusercontent.com/59989652/77994213-30bb7b00-735c-11ea-8458-00dd36495c40.png)